src/error.rs

```diff
@@ -21,6 +30,43 @@ pub type DaftResult<T> = std::result::Result<T, DaftError>;
 impl Display for DaftError {
     // `f` is a buffer, and this method must write the formatted string into it
     fn fmt(&self, f: &mut Formatter) -> Result {
-        write!(f, "{self:?}")
+        // Override this method if the error propagated should have custom display for better user-level error messages
+        match self {
+            Self::ExprResolveTypeError {
+                expectation,
+                op_display_name,
+                binary_op_display,
+                fields_to_expr,
+            } => {
+                let simple_field_display = match (binary_op_display, &fields_to_expr[..]) {
+                    (Some(op), [(left_field, _), (right_field, _)]) => {
```

Comment:
this logic should already be implemented for binary ops, can't we leverage that

Reply:
I want to print just `{field_name} {op} {field_name}` but the binary_op Display would print `{left_expr} {op} {right_expr}`